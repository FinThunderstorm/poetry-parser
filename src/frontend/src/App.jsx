import React from 'react'
import { Outlet } from 'react-router-dom'
import MainMenu from './components/MainMenu'
import Welcome from './components/Welcome'

const App = () => (
    <>
        <MainMenu />
        <Welcome />
        <Outlet />
    </>
)

export default App
