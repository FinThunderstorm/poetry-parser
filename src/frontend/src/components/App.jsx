import React from 'react'
import { Route, Routes } from 'react-router-dom'

import UploadFile from './UploadFile'
import NotFound from './NotFound'
import PackageList from './PackageList'
import Layout from './Layout'
import Package from './Package'

/** App handles routing in frontend
 */
const App = () => (
    <Routes>
        <Route path="*" element={<Layout />}>
            <Route index element={<PackageList />} />
            <Route path="packages" element={<PackageList />}>
                <Route path=":name" element={<Package />} />
                <Route path="*" element={<NotFound />} />
            </Route>
            <Route path="upload" element={<UploadFile />} />
            <Route path="*" element={<NotFound />} />
        </Route>
    </Routes>
)

export default App
