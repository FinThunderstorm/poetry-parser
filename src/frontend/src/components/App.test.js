/* eslint-disable react/prop-types */
import React from 'react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import { configureStore } from '@reduxjs/toolkit'
import App from './App'

import packagesReducer from '../reducers/packagesSlice'

describe('App', () => {
    const AppWrapper = ({ url }) => {
        const store = configureStore({
            reducer: {
                packages: packagesReducer,
            },
        })
        return (
            <MemoryRouter initialEntries={[url]}>
                <Provider store={store}>
                    <App />
                </Provider>
            </MemoryRouter>
        )
    }

    test('App renderes correctly', () => {
        render(<AppWrapper url="/" />)

        const home = screen.getByText('home', {
            exact: false,
        })
        const title = screen.getByText('Welcome to poetry-parser!', {
            exact: false,
        })
        const info = screen.getByText(
            'Begin explorating your poetry.lock -file by uploading file.',
            {
                exact: false,
            }
        )
        expect(home).toBeDefined()
        expect(title).toBeDefined()
        expect(info).toBeDefined()

        const tree = renderer.create(<AppWrapper url="/" />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
