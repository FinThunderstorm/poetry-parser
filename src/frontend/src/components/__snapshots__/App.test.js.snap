// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renderes correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          poetry-parser
        </div>
        <a
          className="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-1y942vo-MuiButtonBase-root-MuiButton-root"
          href="/"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          Home
        </a>
        <a
          className="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-1y942vo-MuiButtonBase-root-MuiButton-root"
          href="/upload"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          Upload
        </a>
      </div>
    </header>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 css-1p3yn26-MuiPaper-root"
    >
      <div
        className="css-1y48f0p-MuiStack-root"
      >
        <span
          className="MuiTypography-root MuiTypography-button css-1xjj6hs-MuiTypography-root"
        >
          Welcome to poetry-parser!
        </span>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Begin explorating your poetry.lock -file by uploading file.
        </p>
        <a
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
          href="/upload"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          Upload
        </a>
      </div>
    </div>
  </div>
</div>
`;
